<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <title>Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials) - Yuan Tian</title><meta name="Description" content="Autoencoder for integrative analysis of CITE-seq data."><meta property="og:title" content="Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)" />
<meta property="og:description" content="Autoencoder for integrative analysis of CITE-seq data." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" /><meta property="og:image" content="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png"/><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2019-09-14T00:00:00-08:00" />
<meta property="article:modified_time" content="2019-09-14T00:00:00-08:00" /><meta property="og:site_name" content="Yuan Tian" />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png"/>
<meta name="twitter:title" content="Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)"/>
<meta name="twitter:description" content="Autoencoder for integrative analysis of CITE-seq data."/>
<meta name="application-name" content="Yuan Tian">
<meta name="apple-mobile-web-app-title" content="Yuan Tian"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" /><link rel="prev" href="https://naity.github.io/refocus-making-out-of-focus-microscopy-images-in-focus-again/" /><link rel="next" href="https://naity.github.io/integrating-image-and-tabular-data-for-deep-learning/" /><link rel="stylesheet" href="/css/style.min.css"><link rel="preload" href="/lib/fontawesome-free/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"></noscript><link rel="preload" href="/lib/animate/animate.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/animate/animate.min.css"></noscript><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)",
        "inLanguage": "en",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "https:\/\/naity.github.io\/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning\/"
        },"image": [{
                            "@type": "ImageObject",
                            "url": "https:\/\/naity.github.io\/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning\/featured-image.png",
                            "width":  713 ,
                            "height":  611 
                        }],"genre": "posts","keywords": "Deep Learning, Machine Learning, Autoencoder","wordcount":  992 ,
        "url": "https:\/\/naity.github.io\/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning\/","datePublished": "2019-09-14T00:00:00-08:00","dateModified": "2019-09-14T00:00:00-08:00","license": "Yuan Tian","publisher": {
            "@type": "Organization",
            "name": ""},"author": {
                "@type": "Person",
                "name": "Yuan Tian"
            },"description": "Autoencoder for integrative analysis of CITE-seq data."
    }
    </script></head>
    <body data-header-desktop="fixed" data-header-mobile="auto"><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('auto' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : 'auto' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="Yuan Tian">Yuan Tian</a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/"> Posts </a><a class="menu-item" href="/publications/"> Publications </a><a class="menu-item" href="/photos/"> Photography </a><a class="menu-item" href="/tags/"> Tags </a><a class="menu-item" href="/categories/"> Categories </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="Search titles or contents..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="Search">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="Clear">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="Switch Theme">
                    <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
                </a></div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="Yuan Tian">Yuan Tian</a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="Search titles or contents..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="Search">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="Clear">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        Cancel
                    </a>
                </div><a class="menu-item" href="/posts/" title="">Posts</a><a class="menu-item" href="/publications/" title="">Publications</a><a class="menu-item" href="/photos/" title="">Photography</a><a class="menu-item" href="/tags/" title="">Tags</a><a class="menu-item" href="/categories/" title="">Categories</a><a href="javascript:void(0);" class="menu-item theme-switch" title="Switch Theme">
                <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
            </a></div>
    </div>
</header><div class="search-dropdown desktop">
        <div id="search-dropdown-desktop"></div>
    </div>
    <div class="search-dropdown mobile">
        <div id="search-dropdown-mobile"></div>
    </div><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">Contents</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animate__animated animate__flipInX">Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="https://www.ytiancompbio.com" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw" aria-hidden="true"></i>Yuan Tian</a></span>&nbsp;<span class="post-category">included in <a href="/categories/machine-learning/"><i class="far fa-folder fa-fw" aria-hidden="true"></i>Machine Learning</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw" aria-hidden="true"></i>&nbsp;<time datetime="09/14/2019">09/14/2019</time>&nbsp;<i class="fas fa-pencil-alt fa-fw" aria-hidden="true"></i>&nbsp;992 words&nbsp;
                <i class="far fa-clock fa-fw" aria-hidden="true"></i>&nbsp;5 minutes&nbsp;</div>
        </div><div class="featured-image"><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png"
        data-srcset="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png 1.5x, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png 2x"
        data-sizes="auto"
        alt="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/featured-image.png"
        title="Autoencoder for integrative analysis of CITE-seq data." /></div><div class="details toc" id="toc-static"  data-kept="">
                <div class="details-summary toc-title">
                    <span>Contents</span>
                    <span><i class="details-icon fas fa-angle-right" aria-hidden="true"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#video-tutorial-1">Video tutorial #1:</a></li>
    <li><a href="#video-tutorial-2">Video tutorial #2:</a></li>
    <li><a href="#data">Data</a></li>
    <li><a href="#autoencoder">Autoencoder</a></li>
    <li><a href="#implementation">Implementation</a></li>
    <li><a href="#results">Results</a></li>
    <li><a href="#summary-and-future-directions">Summary and Future Directions</a></li>
    <li><a href="#source-code">Source Code</a></li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h2 id="video-tutorial-1">Video tutorial #1:</h2>

<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
  <iframe src="https://www.youtube-nocookie.com/embed/tad9TPCMWbU" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" allowfullscreen title="YouTube Video"></iframe>
</div>

<h2 id="video-tutorial-2">Video tutorial #2:</h2>

<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
  <iframe src="https://www.youtube-nocookie.com/embed/sRpj9MhNjS8" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" allowfullscreen title="YouTube Video"></iframe>
</div>

<p>        <font size="6">S</font><font size="4">ingle-cell RNA sequencing (scRNA-seq) has offered a comprehensive and unbiased approach to profile various type of cells such as immune cells with a single-cell resolution using next‑generation sequencing. More recently, exciting technologies such as cellular indexing of transcriptomes and epitopes by sequencing (CITE-seq) have been developed to extend scRNA-seq by jointly measuring multiple molecular modalities such as proteome and transcriptome from the same cell as illustrated in the figure below. By utilizing antibodies that are conjugated to oligonucleotides, CITE-seq simultaneously generates sequencing-based readouts for surface protein expression along with gene expression.</font></p>
<p>Since gene and protein expressions convey distinct and complementary information about a cell, CITE-seq offers a unique opportunity to combine both transcriptomic and proteomic data to decipher the biology of individual cells at a considerably higher resolution than using either one alone. This requires computational methods that can effectively integrate single-cell data from both modalities. In this tutorial, we will conduct integrative analysis of CITE-seq data using an unsupervised deep learning method named autoencoder.</p>
<div class="details admonition note open">
        <div class="details-summary admonition-title">
            <i class="icon fas fa-pencil-alt fa-fw" aria-hidden="true"></i>In essence:<i class="details-icon fas fa-angle-right fa-fw" aria-hidden="true"></i>
        </div>
        <div class="details-content">
            <div class="admonition-content"><ul>
<li>Single-cell technologies offer considerable promise in dissecting the heterogeneity among individual cells and are being utilized in biomedical studies at an astounding pace.</li>
<li>CITE-seq simultaneously measures gene expression and surface protein at a single-cell level.</li>
</ul>
</div>
        </div>
    </div>
<p><figure><a class="lightgallery" href="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg" title="CITE-seq" data-thumbnail="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg" data-sub-html="<h2>Image source: 10x Genomics</h2><p>CITE-seq</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg"
            data-srcset="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg 1.5x, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg 2x"
            data-sizes="auto"
            alt="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/citeseq.jpg" width="546" height="414" />
    </a><figcaption class="image-caption">Image source: 10x Genomics</figcaption>
    </figure></p>
<h2 id="data">Data</h2>
<p>We will be using the first CITE-seq dataset published by Stoeckius M et al. in 2017. The authors measured the single-cell transcriptomics of ~8,000 cord blood mononuclear cells (CBMCs) together with the expression of 13 proteins. There are two CSV files — one for gene expression and the other one for protein expression, which can be downloaded from here. Seurat is an R package designed for QC, analysis, and exploration of single-cell genomics data. We will first preprocess the data before feeding them into our autoencoder and to perform cell clustering and annotation based on gene expression using the Seurat package. The code can be found in this accompanying notebook.</p>
<p>After preprocessing, gene and protein expression data are concatenated together, where each column is a gene or protein while each row is a cell (each cell has a unique barcode). The dataset contains the expression levels of 2000 genes and 10 proteins (3 were removed due to poor enrichment) for a total of 7895 cells.</p>
<script type="application/javascript" src="https://gist.github.com/naity/10951c280955a992a23899a5f5806a83.js"></script>

<h2 id="autoencoder">Autoencoder</h2>
<p>Autoencoder is a type of unsupervised deep learning model or neural network that consists of three major components: an encoder, a bottleneck, and a decoder as shown in the figure below. The encoder compresses the input, and the bottleneck layer stores the compressed representation of the input. In contrast, the decoder tries to reconstruct the input based upon the compressed data.</p>
<p>The dimension of the bottleneck layer is normally substantially lower than that of the input. As a result, the encoder will try to learn as much meaningful information about the input as possible while ignoring the noise so that the decoder can do a better job reconstructing the input. Autoencoder can function as a dimensionality reduction algorithm and the low-dimensional representation of the input stored in the bottleneck layer can be used for data visualization and other purposes. Moreover, thanks to its flexible neural network architecture, it offers unlimited ways to incorporate gene and protein expression data as we shall see below.</p>
<p><figure><a class="lightgallery" href="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png" title="Autoencoder" data-thumbnail="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png" data-sub-html="<h2>Image source: Eraslan et al. Nat Rev Genet. 2019</h2><p>Autoencoder</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png"
            data-srcset="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png 1.5x, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png 2x"
            data-sizes="auto"
            alt="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder.png" width="1725" height="575" />
    </a><figcaption class="image-caption">Image source: Eraslan et al. Nat Rev Genet. 2019</figcaption>
    </figure></p>
<h2 id="implementation">Implementation</h2>
<p>Since gene and protein data have dramatically different dimensions, we will first encode them separately using two different encoders and then concatenate the outputs, which will be passed through another encoder to generate the bottleneck layer. Subsequently, the decoder will try to reconstruct the input based on the bottleneck layer. The overall neural network architecture is illustrated below:</p>
<p><figure><a class="lightgallery" href="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png" title="Autoencoder architecture" data-thumbnail="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png" data-sub-html="<h2>Autoencoder architecture</h2><p>Autoencoder architecture</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png"
            data-srcset="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png 1.5x, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png 2x"
            data-sizes="auto"
            alt="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/autoencoder_arch.png" width="1570" height="714" />
    </a><figcaption class="image-caption">Autoencoder architecture</figcaption>
    </figure></p>
<p>The implementation of the autoencoder is straightforward using Pytorch and fastai libraries. In this example, the dimensions of the compressed gene and protein data are 120 and 8, respectively, and the bottleneck layer consists of 64 neurons. The entire code can be found in <a href="https://github.com/naity/citeseq_autoencoder/blob/master/autoencoder_gene_only.ipynb" target="_blank" rel="noopener noreffer ">this (gene only)</a> and <a href="https://github.com/naity/citeseq_autoencoder/blob/master/autoencoder_gene_protein.ipynb" target="_blank" rel="noopener noreffer ">this (gene and protein)</a> accompanying notebooks.</p>
<script type="application/javascript" src="https://gist.github.com/naity/cc8c8f694fb318fd1a5d1cb60f2b5f8b.js"></script>

<div class="details admonition tip open">
        <div class="details-summary admonition-title">
            <i class="icon fas fa-lightbulb fa-fw" aria-hidden="true"></i>Update<i class="details-icon fas fa-angle-right fa-fw" aria-hidden="true"></i>
        </div>
        <div class="details-content">
            <div class="admonition-content">Updated implementation in Pytorch and <a href="https://www.pytorchlightning.ai/" target="_blank" rel="noopener noreffer ">PyTorch Lightning</a> can be found in this <a href="https://github.com/naity/citeseq_autoencoder/blob/master/autoencoder_citeseq_saturn.ipynb" target="_blank" rel="noopener noreffer ">Jupyter notebook</a>.</div>
        </div>
    </div>
<h2 id="results">Results</h2>
<p>In addition to the model trained on both gene and protein expression data, we also train a control model using gene expression data only. In this way, we would be able to discern any advantages of performing integrative analysis of CITE-seq data. After training the models, we extract the 64-dimensional compressed representation of the original input stored in the bottleneck layer, which is subsequently visualized on a two-dimensional map generated by t-SNE.</p>
<p>As shown in the figures below, the integrative analysis of both transcriptomic and proteomic data generates better results than using gene expression data alone. For example, CD8 T cell (red arrow) and CD4 T cell clusters together form a large “island” and can’t be separated when only gene expression data is used. In contrast, CD8 T cells (red arrow) form a distinct cluster that is well separated from the CD4 T cell clusters when combining protein expression data with gene expression data. Thus, the combinatory analysis is more efficient at analyzing CITE-seq data than using data from a single molecular modality.</p>
<p><figure><a class="lightgallery" href="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg" title="Gene only" data-thumbnail="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg" data-sub-html="<h2>Visualization based on gene expression data only. The red arrow points at CD8 T cells.</h2><p>Gene only</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg"
            data-srcset="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg 1.5x, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg 2x"
            data-sizes="auto"
            alt="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_only.jpeg" width="700" height="388" />
    </a><figcaption class="image-caption">Visualization based on gene expression data only. The red arrow points at CD8 T cells.</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png" title="Gene and protein" data-thumbnail="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png" data-sub-html="<h2>Visualization based on both gene and protein expression data only. The red arrow points at CD8 T cells.</h2><p>Gene and protein</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png"
            data-srcset="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png 1.5x, /integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png 2x"
            data-sizes="auto"
            alt="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/gene_protein.png" width="700" height="397" />
    </a><figcaption class="image-caption">Visualization based on both gene and protein expression data only. The red arrow points at CD8 T cells.</figcaption>
    </figure></p>
<h2 id="summary-and-future-directions">Summary and Future Directions</h2>
<p>Here we built an autoencoder-based deep learning model for dimensionality reduction and visualization of single-cell CITE-seq data. We demonstrate that the integrative analysis of both transcriptomic and proteomic data achieves superior resolution in distinguishing between various immune cell types.</p>
<p>One limitation is that the cells are assigned to different immune cell types during the preprocessing step using the gene expression data alone. It will be of great interest to develop a deep learning model that can also perform cell clustering and characterization using data from multiple molecular modalities.</p>
<h2 id="source-code">Source Code</h2>
<p>Data and Jupyter notebooks have been deposited in this <a href="https://github.com/naity/citeseq_autoencoder" target="_blank" rel="noopener noreffer ">repository</a>.</p></div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>Updated on 09/14/2019</span>
            </div></div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/index.md" target="_blank">Read Markdown</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="Share on Twitter" data-sharer="twitter" data-url="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" data-title="Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)" data-via="_ytian_" data-hashtags="Deep Learning,Machine Learning,Autoencoder"><i class="fab fa-twitter fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Facebook" data-sharer="facebook" data-url="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" data-hashtag="Deep Learning"><i class="fab fa-facebook-square fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Linkedin" data-sharer="linkedin" data-url="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/"><i class="fab fa-linkedin fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Hacker News" data-sharer="hackernews" data-url="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" data-title="Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)"><i class="fab fa-hacker-news fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on Line" data-sharer="line" data-url="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" data-title="Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)"><i data-svg-src="/lib/simple-icons/icons/line.min.svg" aria-hidden="true"></i></a><a href="javascript:void(0);" title="Share on 微博" data-sharer="weibo" data-url="https://naity.github.io/integrative-analysis-of-single-cell-multi-omics-data-using-deep-learning/" data-title="Integrative analysis of single-cell multi-omics data using deep learning (with video tutorials)"><i class="fab fa-weibo fa-fw" aria-hidden="true"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"><i class="fas fa-tags fa-fw" aria-hidden="true"></i>&nbsp;<a href="/tags/deep-learning/">Deep Learning</a>,&nbsp;<a href="/tags/machine-learning/">Machine Learning</a>,&nbsp;<a href="/tags/autoencoder/">Autoencoder</a></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">Back</a></span>&nbsp;|&nbsp;<span><a href="/">Home</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/refocus-making-out-of-focus-microscopy-images-in-focus-again/" class="prev" rel="prev" title="ReFocus: Making Out-of-Focus Microscopy Images In-Focus Again"><i class="fas fa-angle-left fa-fw" aria-hidden="true"></i>ReFocus: Making Out-of-Focus Microscopy Images In-Focus Again</a>
            <a href="/integrating-image-and-tabular-data-for-deep-learning/" class="next" rel="next" title="Integrating image and tabular data for deep learning">Integrating image and tabular data for deep learning<i class="fas fa-angle-right fa-fw" aria-hidden="true"></i></a></div>
</div>
<div id="comments"><div id="disqus_thread" class="comment"></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://disqus.com/?ref_noscript">Disqus</a>.
            </noscript></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">Powered by <a href="https://gohugo.io/" target="_blank" rel="noopener noreffer" title="Hugo 0.109.0">Hugo</a> | Theme - <a href="https://github.com/dillonzq/LoveIt" target="_blank" rel="noopener noreffer" title="LoveIt 0.2.11"><i class="far fa-kiss-wink-heart fa-fw" aria-hidden="true"></i> LoveIt</a>
                </div><div class="footer-line" itemscope itemtype="http://schema.org/CreativeWork"><i class="far fa-copyright fa-fw" aria-hidden="true"></i><span itemprop="copyrightYear">2023</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="www.ytiancompbio.com" target="_blank">Yuan Tian</a></span>&nbsp;|&nbsp;<span class="license"><a rel="license external nofollow noopener noreffer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span></div>
        </div>
    </footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="Back to Top">
                <i class="fas fa-arrow-up fa-fw" aria-hidden="true"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="View Comments">
                <i class="fas fa-comment fa-fw" aria-hidden="true"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/lightgallery/css/lightgallery-bundle.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/cookieconsent/cookieconsent.min.css"><script type="text/javascript" src="https://naity.disqus.com/embed.js" defer></script><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lightgallery.min.js"></script><script type="text/javascript" src="/lib/lightgallery/plugins/thumbnail/lg-thumbnail.min.js"></script><script type="text/javascript" src="/lib/lightgallery/plugins/zoom/lg-zoom.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/mhchem.min.js"></script><script type="text/javascript" src="/lib/cookieconsent/cookieconsent.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"Copy to clipboard","maxShownLines":50},"comment":{},"cookieconsent":{"content":{"dismiss":"Got it!","link":"Learn more","message":"This website uses Cookies to improve your experience."},"enable":true,"palette":{"button":{"background":"#f0f0f0"},"popup":{"background":"#1aa3ff"}},"theme":"edgeless"},"lightgallery":true,"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":true,"left":"\\begin{equation}","right":"\\end{equation}"},{"display":true,"left":"\\begin{equation*}","right":"\\end{equation*}"},{"display":true,"left":"\\begin{align}","right":"\\end{align}"},{"display":true,"left":"\\begin{align*}","right":"\\end{align*}"},{"display":true,"left":"\\begin{alignat}","right":"\\end{alignat}"},{"display":true,"left":"\\begin{alignat*}","right":"\\end{alignat*}"},{"display":true,"left":"\\begin{gather}","right":"\\end{gather}"},{"display":true,"left":"\\begin{CD}","right":"\\end{CD}"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","maxResultLength":10,"noResultsFound":"No results found","snippetLength":30,"type":"lunr"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
